### PR TITLE
204 Add "rendering in progress" info

### LINKED ISSUE #204: [Feature Request]: Add "rendering in progress" info
### Feature description

It might take a while to get rendered report save pop-up window after clicking "download" button. Please add appropriate UI element to inform that rendering is in progress.

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct.

### Contribution Guidelines

- [X] I agree to follow this project's Contribution Guidelines.

### Security Policy

- [X] I agree to follow this project's Security Policy.

### PR DESCRIPTION
**Pull request by chlebowa**
Closes #204.

Added blocking of download buttons with `shinybusy::block`.

#### TESTING APP
```
devtools::load_all()
library(teal.modules.general)

data <- teal_data()
data <- within(data, {
  library(nestcolor)
  ADSL <- teal.modules.general::rADSL
})
datanames <- c("ADSL")
datanames(data) <- datanames
join_keys(data) <- default_cdisc_join_keys[datanames]

app <- teal::init(
  data = data,
  modules = teal::modules(
    teal.modules.general::tm_a_regression(
      label = "Regression",
      response = teal.transform::data_extract_spec(
        dataname = "ADSL",
        select = teal.transform::select_spec(
          label = "Select variable:",
          choices = "BMRKR1",
          selected = "BMRKR1",
          multiple = FALSE,
          fixed = TRUE
        )
      ),
      regressor = teal.transform::data_extract_spec(
        dataname = "ADSL",
        select = teal.transform::select_spec(
          label = "Select variables:",
          choices = teal.transform::variable_choices(data[["ADSL"]], c("AGE", "SEX", "RACE")),
          selected = "AGE",
          multiple = TRUE,
          fixed = FALSE
        )
      ),
      ggplot2_args = teal.widgets::ggplot2_args(
        labs = list(subtitle = "Plot generated by Regression Module")
      )
    )
  )
)
runApp(app, launch.browser = TRUE)
```

### CODE REVIEW

**Comment by Polkas** [comment]
Nice solution; an advantage is that the solution does not add other dependencies than shinybusy to the package.

```r
> tools::package_dependencies("shinybusy")
$shinybusy
[1] "htmltools"   "shiny"       "jsonlite"    "htmlwidgets"
```

**Comment by chlebowa** [comment]
Thanks @Polkas.
It is rather light. 3/4 of its dependencies are already used in `teal.reporter` and `htmlwidgets` bring no new ones. :slightly_smiling_face:

**Comment by pawelru** [comment]
Thanks. I like this. Details to be sorted out but functional wise I feel this is exactly what we need.

**Comment by chlebowa** [comment]
This is what it looks like now:
image

Note that the `Cancel` button merely closes the modal and does not in fact cancel the rendering. If that can be done, it would be a separate issue.

File: R/DownloadModule.R [hunk 1]
```diff
@@ -141,10 +141,13 @@ download_report_button_srv <- function(id,
         },
         content = function(file) {
           shiny::showNotification("Rendering and Downloading the document.")
+          shinybusy::block(id = ns("download_data"), text = "", type = "dots")
+          Sys.sleep(3)
```
Comment (pawelru) [inline R/DownloadModule.R:145]: While it might help with the review - don't forget to revert it before merging!
Pls keep it unresolved as a reminder

Comment (chlebowa) [reply R/DownloadModule.R:145]: Of course :slightly_smiling_face:

**Comment by chlebowa** [comment]
@pawelru Anything else needed here?

**Comment by pawelru** [comment]
Thanks. It's good now. Please remove sys wait and I will be happy to approve.

**Review by pawelru — APPROVED** [review]
👍 thanks!

**Pull request merged**